src/status_im/add_new/core.cljs

```diff
@@ -33,21 +33,26 @@
                               (string/starts-with? new-identity "0x"))
           is-ens? (and (not is-public-key?)
                        (ens/valid-eth-name-prefix? new-identity))
-          error (db/validate-pub-key db new-identity)]
+          error (db/validate-pub-key db new-identity)
+          ens-name (resolver/ens-name-parse new-ens-name)]
       (reset! resolve-last-id nil)
-      (merge {:db (assoc db
-                         :contacts/new-identity
-                         {:public-key new-identity
-                          :state      (cond is-ens?
-                                            :searching
-                                            (and (string/blank? new-identity) (not new-ens-name))
-                                            :empty
-                                            error
-                                            :error
-                                            :else
-                                            :valid)
-                          :error      error
-                          :ens-name   (resolver/ens-name-parse new-ens-name)})}
+      (merge {:db (as-> db $
```

Comment:
could you please elaborate on using `as->`. here ? thanks

Reply:
Was to add an if statement inside the macro, but I'm pretty sure there are better ways to do it

Reply:
`cond->` ?

Reply:
That looks way cleaner :)